docs/rules/comma-dangle.md

```diff
@@ -39,13 +39,40 @@ This rule enforces consistent use of trailing commas in object and array literal
 
 ## Options
 
-This rule has a string option:
+This rule has a string option or an object option:
+
+```json
+{
+    "comma-dangle": ["error", "never"],
+    // or
+    "comma-dangle": ["error", {
+        "arrays": "never",
+        "objects": "never",
+        "imports": "never",
+        "exports": "never",
+        "functions": "ignore",
+    }]
+}
+```
 
 * `"never"` (default) disallows trailing commas
 * `"always"` requires trailing commas
 * `"always-multiline"` requires trailing commas when the last element or property is in a *different* line than the closing `]` or `}` and disallows trailing commas when the last element or property is on the *same* line as the closing `]` or `}`
 * `"only-multiline"` allows (but does not require) trailing commas when the last element or property is in a *different* line than the closing `]` or `}` and disallows trailing commas when the last element or property is on the *same* line as the closing `]` or `}`
 
+Trailing commas in function declarations and function calls are valid syntax since EcmaScript® 2017. So the string option does not check function declarations and function calls for backward compatibility.
```

Comment:
Very minor nit: I think there should be a semicolon after "2017".

> Trailing commas in function declarations and function calls are valid syntax since ECMAScript 2017; however, the string option does not check these situations for backwards compatibility.